web/src/attachments_ui.ts

```diff
@@ -14,31 +15,63 @@ import * as scroll_util from "./scroll_util";
 import * as timerender from "./timerender";
 import * as ui_report from "./ui_report";
 
-let attachments;
-let upload_space_used;
-
-export function bytes_to_size(bytes, kb_with_1024_bytes = false) {
+type ServerAttachment = z.infer<typeof attachment_api_response_schema>["attachments"][number];
+
+type Attachment = ServerAttachment & {
+    create_time_str: string;
+    size_str: string;
+};
+
+type AttachmentEvent = {
+    op: "add" | "remove" | "update";
```

Comment:
For op "remove", `attachment` would be `{id: number}`.
https://github.com/zulip/zulip/blob/81bd63cb46273b8c94ef9e92c00893ed97110119/zerver/openapi/zulip.yaml#L1542-L1582